File: app/src/main/java/com/vn/qrscanner/Utilities/PermissionUtil.java
package com.vn.qrscanner.Utilities;

import android.content.Context;
import android.content.pm.PackageManager;
import android.os.Build;
import androidx.appcompat.app.AppCompatActivity;

import java.util.ArrayList;

public class PermissionUtil {


    public static boolean verifyPermission(Context context, String permission) {
        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) return true;
        return context.checkSelfPermission(permission) == PackageManager.PERMISSION_GRANTED;
    }


    public static void requestPermissions(AppCompatActivity activity, String[] permissions, int requestCode) {
        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) return;

        ArrayList<String> missingPermissions = new ArrayList<>();
        for (String permission : permissions) {
            if (!verifyPermission(activity, permission)) {
                missingPermissions.add(permission);
            }
        }
        if (missingPermissions.size() != 0)
            activity.requestPermissions(missingPermissions.toArray(new String[missingPermissions.size()]), requestCode);
    }
}

File: app/src/main/java/com/vn/qrscanner/utils/PermissionUtil.kt
package com.vn.qrscanner.utils

import android.content.Context
import android.content.pm.PackageManager
import android.os.Build
import androidx.appcompat.app.AppCompatActivity
import java.util.ArrayList

object PermissionUtil {
    fun verifyPermission(context: Context, permission: String?): Boolean {
        return context.checkSelfPermission(permission!!) == PackageManager.PERMISSION_GRANTED
    }

    fun requestPermissions(activity: AppCompatActivity, permissions: Array<String>, requestCode: Int) {
        val missingPermissions = ArrayList<String>()
        for (permission in permissions) {
            if (!verifyPermission(activity, permission)) {
                missingPermissions.add(permission)
            }
        }
        if (missingPermissions.size != 0) activity.requestPermissions(missingPermissions.toTypedArray(), requestCode)
    }
}